components/scream/src/physics/zm/atmosphere_deepconvection.cpp

```diff
@@ -0,0 +1,165 @@
+#include "physics/zm/scream_zm_interface.hpp"
+#include "physics/zm/atmosphere_deepconvection.hpp"
+#include "physics/zm/zm_inputs_initializer.cpp"
+
+namespace scream
+
+{
+
+ZMDeepConvection::ZMDeepConvection (const Comm& comm,const ParameterList& /* params */)
+  : m_zm_comm (comm)
+{
+  m_initializer = create_field_initializer<ZMInputsInitializer>();
+  
+}
+
+
+void ZMDeepConvection::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace std;
+
+  using namespace units;
+  auto Q = kg/kg;
+  auto nondim = m/m;
```

Comment:
this must be C++ magic I don't understand: how can Q=kg/kg be a valid variable assignment? I would think you'd need quotes around kg/kg and m/m.

Reply:
This is totally C++ magic.  It has to do with the namespace units that is being used and it is something that Luca put in to deal with the mass mixing ratio units issue where the units are kg_X/kg_air, which is technically kg/kg.

Reply:
`auto Q=kg/kg;` is a normal assignment, just like `auto x = 1.0 / 3.0;`. Right after the assignment, however, Q is a nondimensional unit (in fact, Q and nondim are the same). The following line is what causes Q to print a user defined string when outputed to screen (here "kg/kg"), while nondim would show "[1]" (or "1", I don't recall).

The `set_string` method accepts anything. One could do `Q.set_string("Luca is the best");`. There's no check on what string is passed. Since this is only used at "cout" time, it's not a big deal though if two places use two different strings (like "kg/kg" and "kg  / kg ").

Reply:
Ok, that makes sense. Thanks Luca!

Reply:
@PeterCaldwell sorry, I put the second comment in pretty much right after the first, but I started a review (rather than post single comment) that I didn't complete, so It didn't posted it until now (when I noticed). I hope it now makes more sense.